visualization/app/codeCharta/util/scenarioHelper.ts

```diff
@@ -1,41 +1,254 @@
+						margin: dynamicSettings.margin
+					}
+					break
+				}
+				case "heightMetric": {
+					scenario.height = {
+						heightMetric: dynamicSettings.heightMetric,
+						labelSlider: appSettings.amountOfTopLabels,
+						heightSlider: appSettings.scaling
+					}
+					break
+				}
+				case "colorMetric": {
+					scenario.color = {
+						colorMetric: dynamicSettings.colorMetric,
+						colorRange: dynamicSettings.colorRange
+					}
+					break
+				}
+				case "edgeMetric": {
+					scenario.edge = {
+						edgeMetric: dynamicSettings.edgeMetric,
+						edgeHeight: appSettings.edgeHeight,
+						edgePreview: appSettings.amountOfEdgePreviews
+					}
+					break
+				}
+			}
+			if (appSettings.camera) {
+				scenario.camera = {
+					camera: appSettings.camera,
+					cameraTarget: appSettings.cameraTarget
+				}
+			}
+		}
+
+		return scenario
+	}
+
+	private static setScenariosToLocalStorage(scenarios: RecursivePartial<Scenario>[]) {
+		const newLocalStorageElement: CCLocalStorage = { version: this.CC_LOCAL_STORAGE_VERSION, scenarios }
+		localStorage.setItem("scenarios", JSON.stringify(newLocalStorageElement))
+	}
+
+	private static loadScenarios(): RecursivePartial<Scenario>[] {
+		const ccLocalStorage: CCLocalStorage = JSON.parse(localStorage.getItem("scenarios"))
+		if (ccLocalStorage) {
+			return ccLocalStorage.scenarios
+		} else {
+			this.setScenariosToLocalStorage(this.getPreLoadScenarios())
+			return this.getPreLoadScenarios()
+		}
+	}
+
+	public static addScenario(newScenario: RecursivePartial<Scenario>) {
+		this.scenarioList.push(newScenario)
+		this.setScenariosToLocalStorage(this.scenarioList)
+	}
+
+	public static createNewScenario(scenarioName: string, scenarioAttributes: AddScenarioContent[]) {
+		const newScenario: RecursivePartial<Scenario> = { name: scenarioName }
+
+		scenarioAttributes.forEach(attribute => {
+			switch (attribute.metricType) {
+				case ScenarioMetricType.CAMERA_POSITION: {
+					newScenario.camera = {
+						camera: attribute.savedValues["camera"],
+						cameraTarget: attribute.savedValues["cameraTarget"]
+					}
+					break
+				}
+				case ScenarioMetricType.AREA_METRIC: {
+					newScenario.area = {
+						areaMetric: attribute.metricName,
+						margin: attribute.savedValues
+					}
+					break
+				}
+				case ScenarioMetricType.HEIGHT_METRIC: {
+					newScenario.height = {
+						heightMetric: attribute.metricName,
+						heightSlider: attribute.savedValues["heightSlider"],
+						labelSlider: attribute.savedValues["labelSlider"]
+					}
+					break
+				}
+				case ScenarioMetricType.COLOR_METRIC: {
+					newScenario.color = {
+						colorMetric: attribute.metricName,
+						colorRange: attribute.savedValues
+					}
+					break
+				}
+				case ScenarioMetricType.EDGE_METRIC: {
+					newScenario.edge = {
+						edgeMetric: attribute.metricName,
+						edgePreview: attribute.savedValues["edgePreview"],
+						edgeHeight: attribute.savedValues["edgeHeight"]
```

Comment:
Would it be possible to just unify the `savedValues` format in a way that all properties of that are just copied 1-to-1 and the metricName is optional? It might be possible to reduce the required transformations quite a bit that way (if that's possible).